src/main/scala/io/iohk/ethereum/db/dataSource/IodbDataSource.scala

```diff
@@ -32,4 +53,34 @@ object IodbDataSource {
   private def storageVersionGen(): Array[Byte] = {
     ByteBuffer.allocate(java.lang.Long.SIZE / java.lang.Byte.SIZE).putLong(updateCounter.incrementAndGet()).array()
   }
+
+  /**
+    * This function constructs an IodbDataSource.
+    *
+    * @param path of the folder where the DataSource files will be stored.
+    * @param keySize of the keys to be stored in the DataSource.
+    *                This length includes the length of the namespace and the length of the keys inside this namespace
+    * @param recreate boolean that, if set to true, this function will return a DataSource with no data in it.
+    *                 If set to false the returned DataSource will continue to use previously built DataSource files.
+    * @return an IodbDataSource.
+    */
+  def apply(path: String, keySize: Int, recreate: Boolean): IodbDataSource = {
+    val dir: File = new File(path)
+    val dirSetupSuccess =
+      if(recreate) (!dir.exists() || deleteDirectory(dir)) && dir.mkdirs()
+      else dir.exists() && dir.isDirectory
+    assert(dirSetupSuccess, "Iodb folder creation failed")
+
+    val lSMStore: LSMStore = new LSMStore(dir = dir, keySize = keySize, keepSingleVersion = true)
+    new IodbDataSource(lSMStore, keySize, path)
+  }
+
+  private def deleteDirectory(dir: File): Boolean = {
+    require(dir.exists() && dir.isDirectory, "Trying to delete a file thats not a folder")
+    val files = Option(dir.listFiles()).getOrElse(Array())
+    val filesDeletionSuccess: Boolean = files.map { f =>
```

Comment:
I agree with stopping at first error but instead of checking if the result of `allFiles.find` is empty I used `forall` which also stops at first error. I also kept separate `dir.delete()` because if not a `StackOverflowError` was thrown due to infinite recursion.